### PR TITLE
scripts: set_assignee: support areas with multiple maintainers

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Rework the set_assignees script to support ares with multiple maintainers. Internally key the maint dictionary by the maintainer set instead of the individual maintaine and then process the individual usernames at the end, assigning to all of them rather than only to the first one.

The rest of the logic is unchanged, but the special case of submitter == maintainer is removed, as that does not seem to be relevant anyway.

Example run:

```
$ ./scripts/set_assignees.py -v -y -P 54286
...
candidate maintainers: {('fabiobaltieri', 'keith-zephyr'): 7}
Picked maintainer: ('fabiobaltieri', 'keith-zephyr') (50.00% ownership)
...
Adding assignee NamedUser(login="fabiobaltieri")...
Adding assignee NamedUser(login="keith-zephyr")...
```

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Followup from https://github.com/zephyrproject-rtos/zephyr/issues/60316

**Comment by fabiobaltieri** [comment]
Hey @nashif we talked about this at dev meeting, were thinking, how about assigning a random person from areas with multiple maintainers instead?

**Comment by nashif** [comment]
> Hey @nashif we talked about this at dev meeting, were thinking, how about assigning a random person from areas with multiple maintainers instead?

not sure what you mean..

**Comment by fabiobaltieri** [comment]
> not sure what you mean..

If the script finds an area with more than one maintainer, instead of always assigning the RP to the first one in the list (what it does now), assign to a random one from the area

**Comment by nashif** [comment]
> If the script finds an area with more than one maintainer, instead of always assigning the RP to the first one in the list (what it does now), assign to a random one from the area

sure, but that is not the issue, I was referring above to issues with multiple lables of multiple areas, which will end up in the script assigning to multiple maintainers of multiple areas. 

Bottom like is, we need to try and  end up with 1 assignee per issue.

**Comment by fabiobaltieri** [comment]
> > If the script finds an area with more than one maintainer, instead of always assigning the RP to the first one in the list (what it does now), assign to a random one from the area
> 
> sure, but that is not the issue, I was referring above to issues with multiple lables of multiple areas, which will end up in the script assigning to multiple maintainers of multiple areas.
> 
> Bottom like is, we need to try and end up with 1 assignee per issue.

That was https://github.com/zephyrproject-rtos/zephyr/pull/61513, this is the one where I was changing how to handle PR assignment.

**Comment by nashif** [comment]
> That was #61513, the one with the "assign issue" feature, this is the one where I was changing how to handle PR assignment. I also got the multi-label story story sorted btw, see [#61513 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/61513#issuecomment-1690378728).

oops, too many PR, got me confused. Random is ok and (not submitter if submitter is maintainer, which we have now)

**Comment by nashif** [comment]
> The rest of the logic is unchanged, but the special case of submitter == maintainer is removed, as that does not seem to be relevant anyway.

why is this not relevant? 

If A and B are maintainers, A submits a PR, it should be assigned to B. This is some we discussed and we agreed to follow.

File: scripts/set_assignees.py [hunk 1]
```diff
@@ -119,19 +118,6 @@ def process_pr(gh, maintainer_file, number):
                                 maintainer = a.maintainers[0]
                                 break
 
-
-        # if the submitter is the same as the maintainer, check if we have
```
Comment (nashif) [inline scripts/set_assignees.py:123]: This should stay.

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by fabiobaltieri** [comment]
> > The rest of the logic is unchanged, but the special case of submitter == maintainer is removed, as that does not seem to be relevant anyway.
> 
> why is this not relevant?
> 
> If A and B are maintainers, A submits a PR, it should be assigned to B. This is some we discussed and we agreed to follow.

Because I was changing the behavior to assign to all maintainers in the area anyway, so that made no difference.

Anyway, I was reworking this and I think this change would hit more corner cases, need to think more about this and I'd also like to restructure the file a bit to make it easier to understand. That means the scope would change anyway, I'll close this down and think the whole story through for a bit longer and probably reopen a new PR.

Thanks for the feedback!

**Pull request closed without merging**